swirl_proc_macro/src/diagnostic_shim.rs

```diff
@@ -0,0 +1,57 @@
+use proc_macro2::{Span, TokenStream};
```

Comment:
That module should probably be upstream into `proc_macro2` because we use it also in diesel and it seems to be useful for other proc macro crates too.